highfidelity/hifi
# Android: Local texture compression using Etc2Comp

https://highfidelity.manuscript.com/f/cases/14785/Compression-of-texture-for-Mobile-format

Test plan (Android):
- We need to do some testing to determine if this is actually beneficial or not.  Run the following steps on multiple phones, once using this PR, and once using the latest from https://github.com/highfidelity/hifi/pull/13152.  Between tests, **make sure to completely uninstall Interface from the phone**.
- Go to a domain (dev-mobile, dev-mobile-master, and Eschatology, etc.).  Don't move or rotate.  Bring up the stats and watch the bottom two rows of the middle expanded column.  The top line is downloads and the bottom line is textures.  Record how long it takes for all of the numbers to go to 0 and stay there.  Record your game and render rates.
- Start a new timer and look all the way around.  Continue rotating your view until all 4 of the numbers go to 0 again.  Record that time.  Close Interface.
- Go into your device settings and find the memory usage for Interface.  It's in a different place on different phones.  Record the memory usage.
- Uninstall interface or press "Clear Data" in Storage.
- Repeat the steps on a few domains.
- In each domain, also make sure that textures look correct (possibly with more artifacts than on desktop).  Make sure that the skybox looks correct (possibly lower resolution), and on domains with irradiance like dev-mobile, make sure the lighting looks correct.

Test plan (Desktop):
- Open Interface, click Developer -> Render -> Clear KTX Cache.  Restart Interface.
- Make sure textures look correct in a few domains.
